tower-batch/src/semaphore.rs

```diff
@@ -0,0 +1,78 @@
+// Copied from tower/src/semaphore.rs
+// When/if tower-batch is upstreamed, delete this file
+// and use the common tower semaphore implementation
```

Comment:
Is it worth opening a ticket for this TODO?

Reply:
I don't think so, because upstreaming this code isn't a work-item for Zebra.  I can do it whenever there's time after we're functionally complete, and we've proven the design (by using it for batch verification).